docs/orchestrating-elastic-stack-applications/beat.asciidoc

```diff
@@ -0,0 +1,378 @@
+
+[id="{p}-beat-configuration"]
+== Configuration
+
+[id="{p}-beat-upgrade-specification"]
+=== Upgrade the Beat specification
+
+Any setting can be changed in the Beat YAML specification, including version upgrades. ECK detects those changes and ensures a smooth rolling upgrade of all Beat Pods. Depending on specification settings used, ECK will set the <<{p}-beat-set-beat-output,output>> part of the config, perform Kibana dashboard <<{p}-beat-set-up-kibana-dashboards,setup>>, restart Beats on certificates rollover and set up the Beats <<{p}-beat-secrets-keystore-for-secure-settings,keystore>>.
+
+[id="{p}-beat-custom-configuration"]
+=== Customize Beat configuration
+
+The Beat configuration is provided through the `config` element:
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: quickstart
+spec:
+  type: heartbeat
+  version: {version}
+  elasticsearchRef:
+    name: quickstart
+  config:
+    heartbeat.monitors:
+    - type: tcp
+      schedule: '@every 5s'
+      hosts: ["quickstart-es-http.default.svc:9200"]
+  deployment:
+    podTemplate:
+      spec:
+        dnsPolicy: ClusterFirstWithHostNet
+        securityContext:
+          runAsUser: 0
+----
+
+Alternatively, it can be provided via a Secret specified in the `configRef` element:
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: heartbeat-quickstart
+spec:
+  type: heartbeat
+  version: 7.8.0
+  elasticsearchRef:
+    name: quickstart
+  configRef:
+    secretName: heartbeat-config
+  deployment:
+    podTemplate:
+      spec:
+        dnsPolicy: ClusterFirstWithHostNet
+        securityContext:
+          runAsUser: 0
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: heartbeat-config
+stringData:
+  beat.yml: |-
+    heartbeat.monitors:
+    - type: tcp
+      schedule: '@every 5s'
+      hosts: ["quickstart-es-http.default.svc:9200"]
+----
+
+For more details about Beats configuration, see the link:https://www.elastic.co/guide/en/beats/libbeat/current/beats-reference.html[Beats documentation].
+
+[id="{p}-beat-deploy-elastic-beat"]
+=== Deploy a Beat
+
+ECK supports the deployment of the following Beats:
+
+- link:https://www.elastic.co/beats/filebeat[Filebeat]
+- link:https://www.elastic.co/beats/metricbeat[Metricbeat]
+- link:https://www.elastic.co/beats/heartbeat[Heartbeat]
+- link:https://www.elastic.co/beats/auditbeat[Auditbeat]
+- link:https://www.elastic.co/beats/packetbeat[Packetbeat]
+- link:https://www.elastic.co/guide/en/beats/journalbeat/current/index.html[Journalbeat]
+
+You can specify the Beat to deploy and its version through `type` and `version` elements. ECK computes the correct path to the container image, so the `image` element doesn't have to be provided. ECK creates a new user in Elasticsearch with a minimal set of appropriate roles and permissions to enable the use of all Beats features.
```

Comment:
```suggestion
You can specify the Beat to deploy and its version through `type` and `version` elements. ECK computes the correct path to the container image, so the `image` element can be omitted. ECK creates a new user in Elasticsearch with a minimal set of appropriate roles and permissions to enable the use of all Beats features.
```

though nowhere in the docs do we even reference the `image` field before now so I wonder if this is even necessary here. Maybe it can just be mentioned as a clarification in the next community beat section?